src/Compilers/CSharp/Portable/Compilation/AttributeSemanticModel.cs

```diff
@@ -48,7 +50,7 @@ public static AttributeSemanticModel CreateSpeculative(SyntaxTreeSemanticModel p
             Debug.Assert(parentSemanticModel != null);
             Debug.Assert(rootBinder != null);
             Debug.Assert(rootBinder.IsSemanticModelBinder);
-
+            rootBinder = new ExecutableCodeBinder(syntax, rootBinder.ContainingMember(), rootBinder);
```

Comment:
>rootBinder [](http://example.com/codeflow?start=12&length=10)

:memo: We should discuss whether a contextual attribute binder should be added here too.

Reply:
I've added the following scenarios as discussed:

```
[My(a)]
void M<T>() { }
```

Scenario 2A:
`GetSpeculativeTypeInfo` for `nameof(TParameter)` and `TParameter` expressions (this works in C# 11)

Scenario 2B:
`TryGetSpeculativeSemanticModel` with a new attribute (this still wouldn't work because replacing an attribute with a new attribute means we're not inside an AttributeSemanticModel, so get can't get a useful target for the new attribute)

Reply:
> I've added the following scenarios as discussed:

Could you point me to the tests?

Reply:
`TypeParameterScope_InMethodAttributeNameOf_SpeculatingWithNewAttribute` is "scenario 1".
`TypeParameterScope_InMethodAttributeNameOf_SpeculatingWithinAttribute` is "2A".
`TypeParameterScope_InMethodAttributeNameOf_SpeculatingWithReplacementAttribute` is "2B".